src/iso15118/d20/state/dc_charge_parameter_discovery.cpp

```diff
@@ -42,6 +42,81 @@ template <> void convert(BPT_DC_ModeRes& out, const d20::DcTransferLimits& in) {
     }
 }
 
+bool handle_compatibility_check(const d20::DcTransferLimits& evse_dc_limits, const session::feedback::DcMaximumLimits ev_max_limits) {
+    // In IEC 61851-23-3 a compatibility check is required
+    // There are voltage, current and power values indexed with CPD and RATED.
+    // The CPD is the communicated value in DC_ChargeParameterDiscoveryRes and RATED the value coming from
+    // the EVerest environment (power supply, grid), here we assume that both are the same
+
+    bool compatiblity_flag = true;
+    const auto ev_max_limits_cpd = ev_max_limits;
+    const float evse_voltage_max_cpd = dt::from_RationalNumber(evse_dc_limits.voltage.max);
+    const float evse_voltage_min_cpd = dt::from_RationalNumber(evse_dc_limits.voltage.min);
+    const float evse_current_max_cpd = dt::from_RationalNumber(evse_dc_limits.charge_limits.current.max);
+    const float evse_current_min_cpd = dt::from_RationalNumber(evse_dc_limits.charge_limits.current.min);
+    const float evse_power_max_cpd = dt::from_RationalNumber(evse_dc_limits.charge_limits.power.max);
+    const float evse_power_min_cpd = dt::from_RationalNumber(evse_dc_limits.charge_limits.power.min);
+    const float evse_voltage_max_rated = evse_voltage_max_cpd;
+    const float evse_current_max_rated = evse_current_max_cpd;
+    const float evse_power_max_rated = evse_power_max_cpd;
+
+    // CC.5.6 2.d-f
+    const float evse_voltage_min_rated = evse_voltage_min_cpd;
+    const float evse_current_min_rated = evse_current_min_cpd;
+    const float evse_power_min_rated = evse_power_min_cpd;
+
+
+    // CC.5.6 2.a
+    if (ev_max_limits_cpd.voltage <= 500.f) {
```

Comment:
Too many magic numbers. Add constexpr for every magic number within the function